python/tank/authentication/login_dialog.py

```diff
@@ -52,6 +54,18 @@
 THREAD_WAIT_TIMEOUT_MS = 5000
 
 
+def _is_context_desktop():
+    """
+    Indicate if we are in the context of the ShotGrid Desktop.
+
+    When the ShotGrid Desktop is used, we want to disregard the value returned
+    by the call to `get_shotgun_authenticator_support_web_login()` when the
+    target site is using Autodesk Identity.
+    """
+    executable_name = os.path.splitext(os.path.basename(sys.executable))[0].lower()
+    return executable_name == "shotgun"
```

Comment:
This code is going to break on Windows, since the executable is named `shotgun.exe`. This is why the `configuration_writer.py` uses `os.path.split(executable)[1].lower().startswith("shotgun")`.

Reply:
I believe that you did not look closely at the code: I use `os.path.splitext` NOT `os.path.split`
I tested locally on my machine, renaming the Mac OS binary to `Shotgun.exe`, and it was detected as being the desktop.

Actually, I have used this setup:

    % ls -al /Applications/Shotgun.app/Contents/MacOS
    total 216
    drwxr-xr-x 5 root wheel   160 Jul 11 19:27 ./
    drwxr-xr-x 7 root wheel   224 Jun  4 09:35 ../
    -rwxr-xr-x 1 root wheel 71744 Jun  4 09:42 Shotgun
    -rwxr-xr-x 1 root wheel 71744 Jul 11 19:25 Shotgun.exe
    -rwxr-xr-x 1 root wheel 71744 Jul 11 19:27 ShotgunXXX.exe

And both `Shotgun` and `Shotgun.exe` where detected as being `_is_running_in_desktop() == True`
With `ShotgunXXX.exe` and using a `.py` file, it was properly deemed as on being in the desktop.

Reply:
![image](https://user-images.githubusercontent.com/8126447/125624987-daf3b640-01f6-4133-b44b-6ffee712a588.png)
You're right. I missed the `splitext`.

Could you test for `shotgun` and `shotgrid` please?